frontend/appflowy_flutter/lib/plugins/database/widgets/group/database_group.dart

```diff
@@ -150,19 +193,31 @@ class _GridGroupCell extends StatelessWidget {
         child: FlowyButton(
           hoverColor: AFThemeExtension.of(context).lightGreyHover,
           text: FlowyText.medium(
-            fieldInfo.name,
+            name,
             color: AFThemeExtension.of(context).textColor,
           ),
-          leftIcon: FlowySvg(
-            fieldInfo.fieldType.svgData,
-            color: Theme.of(context).iconTheme.color,
-          ),
+          leftIcon: icon != null
+              ? FlowySvg(
+                  icon!,
+                  color: Theme.of(context).iconTheme.color,
+                )
+              : null,
           rightIcon: rightIcon,
           onTap: () {
+            List<int> settingContent = [];
+            switch (fieldInfo.fieldType) {
+              case FieldType.DateTime:
+                final config = DateGroupConfigurationPB()
+                  ..condition = DateConditionPB.values[condition];
+                settingContent = config.writeToBuffer();
+                break;
+              default:
+            }
             context.read<DatabaseGroupBloc>().add(
                   DatabaseGroupEvent.setGroupByField(
                     fieldInfo.id,
                     fieldInfo.fieldType,
+                    settingContent,
                   ),
                 );
             onSelected();
```

Comment:
What do you think about splitting the functionality of setting which field to set by and then the adjust group condition setting?

Reply:
I didn't get what resolving this meant, Should I split them or not?

Reply:
Notion doesn't preserve group settings.

Reply:
Yep I think it's fine to keep them together for now.